.github/conftest-unified-ci-rules.rego

```diff
@@ -159,3 +169,25 @@ get_jobs_not_needing_detectchanges(jobInput) = jobs_not_needing_detectchanges {
         count(job_needs_detectchanges) == 0
     }
 }
+
+get_jobs_after_checkresults_without_ifalways(jobInput) = jobs_after_checkresults_without_ifalways {
+    jobs_after_checkresults_without_ifalways := { job_id |
+        job := jobInput[job_id]
+
+        # check if job declares dependency on "check-results" job anywhere
+        job_needs_checkresults := { need |
+            need := job.needs[_]
+            need == "check-results"
+        }
+        count(job_needs_checkresults) == 1
+
+        # check that the job declares an `if: ...` condition and it contains `always()`
+        #
+        # this is important because GHA by default skips execution of jobs that itself
+        # depend on jobs (transitively) that were skipped - which happens a lot in Unified CI
+        # and we want to avoid accidentally skipping deploy jobs or similar
+        #
+        job_if := object.get(job, "if", "")  # get with empty default value
+        not contains(job_if, "always()")
```

Comment:
should we also check if it is testing the success of check results? I think that this might be more desirable otherwise the jobs after check results will always run, even when check-results failed. Does this make sense?